sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -303,6 +312,7 @@ protected AmqpConnectionScope()
             bool isSessionReceiver,
             CancellationToken cancellationToken)
         {
+            Argument.AssertNotDisposed(_disposed, nameof(AmqpConnectionScope));
```

Comment:
Do we need these asserts in AmqpConnectionScope? We have the asserts on the public APIs and it is racy anyways as to whether the connection can be closed after this line is hit.

Reply:
Functionally, we don't.  Worst case, we'd hit the error behavior in the AMQP library and then detect the close and opt not to retry.   My intent was to try and offer a more deterministic exception/message for this scenario.   The AMQP library currently can throw a either an `ObjectDisposedException` or `InvalidOperationException` depending on parts of its internal state with no real context for them.  

While this check doesn't eliminate that from happening, it does give us another chance to offer a more consistent and meaningful message.   Unless you're really offended by it, I'd prefer to leave it in.

Reply:
Sounds good to me.